src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -116,29 +112,40 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
   }
   def getSubjectFromEmail(email: WorkbenchEmail, samRequestContext: SamRequestContext): Future[Option[WorkbenchSubject]] = directoryDAO.loadSubjectFromEmail(email, samRequestContext).unsafeToFuture()
 
-  def getUserStatus(userId: WorkbenchUserId, userDetailsOnly: Boolean = false, samRequestContext: SamRequestContext): Future[Option[UserStatus]] =
+  def getUserStatus(userId: WorkbenchUserId, userDetailsOnly: Boolean = false, samRequestContext: SamRequestContext): Future[Option[UserStatus]] = {
     directoryDAO.loadUser(userId, samRequestContext).unsafeToFuture().flatMap {
       case Some(user) =>
-        if(userDetailsOnly)
+        if (userDetailsOnly)
           Future.successful(Option(UserStatus(UserStatusDetails(user.id, user.email), Map.empty)))
         else
           for {
             googleStatus <- cloudExtensions.getUserStatus(user)
             allUsersGroup <- cloudExtensions.getOrCreateAllUsersGroup(directoryDAO, samRequestContext)
             allUsersStatus <- directoryDAO.isGroupMember(allUsersGroup.id, user.id, samRequestContext).unsafeToFuture() recover { case _: NameNotFoundException => false }
+            tosAcceptedStatus <- {
+              tosService.getTosStatus(user.id).unsafeToFuture()
```

Comment:
I saw that unsafeToFuture forces evaluation of the IO at this time, is there a reason to do the evaluation now vs later? In the UserStatusDiagnostics function I see the evaluations were pulled out of the for comprehension for parallelism. Would that make sense to do here with allUsersStatus, tosAcceptedStatus, and ldapStatus?

Reply:
I think it would depend on what's important to us. If made all of these parallel, it might save some time. However, this way I think we decrease the number of calls we're making to the postgres and opendj dbs (if a previous call fails, there's no need to check the rest). I'm going to punt making a decision on parallel vs sequential in this PR though, if that's alright!

Reply:
That makes sense! I think it's always something we could look into if we needed to improve latency for some use-case. And definitely simplifies logic to be sequential-